ion-schema/src/main/kotlin/com/amazon/ionschema/model/AstSchema.kt

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package com.amazon.ionschema.model
+
+import com.amazon.ionelement.api.IonElement
+
+/**
+ * Represents a complete Ion Schema document. Note—the model does not include a particular ISL version because the AST
+ * is intended to be a version-agnostic representation of ISL.
+ */
+data class AstSchema(
+    val id: String? = null,
+    val header: AstHeader,
+    val content: List<SchemaContent>,
+    val footer: AstFooter,
+) {
+
+    val types get() = content.filterIsInstance<SchemaContent.NamedType>()
```

Comment:
It might be surprising for users that a property access does this filtering. If you were to make it lazy then it would only need to filter once.

Reply:
That is true, but then it would also mean allocating a `Lazy` object and duplicating a subset of the original list. It's something worth considering, but I'd rather deal with it later since it's a backwards compatible change in either direction.